keps/sig-api-machinery/20190226-network-proxy.md

```diff
@@ -160,11 +132,24 @@ Implies no overlapping (i.e. shared) IPs on the network.
 
 We will run a connectivity proxy inside the master network.
 It could work on either a HTTP Connect mechanism or gRPC.
-We will test HTTP Connect to determine scalability and error handling.
-If no issues with either multi-plexing (to prevent an proliferation of
-connections) or with handling connection errors and terminations, then we will
-go with HTTP Connect. We expect to handle metadata in this case using http 'X' headers.
-Alternately the connectivity proxy can expose a gRPC interface to the KAS.
+For the alpha version we will attempt to get this working with HTTP Connect.
+We will evaluate HTTP Connect for scalability, error handling and traffic types.
+For scalability we will be looking at the number of required open connections.
+Increasing usage of webhooks means we need better than 1 request per connection (multiplexing).
+We also need the tunnel to be tolerant of errors in the requests it is transporting.
+HTTP-Connect only supports HTTP requests and not things like DNS requests.
+We assume that for HTTP URL request,s it will be the proxy which does the DNS lookup.
+However this means that we cannot have the KAS perform a DNS request to then do a follow on request. 
+If no issues are found with HTTP Connect in these areas we will proceed with it.
+If an issue is found then we will update the KEP and switch the client to the gRPC solution.
+This should be as simple as switching the connection mode of the client code.
+
+It may be desirable to allow out of band data (metadata) to be transmitted from the KAS to the Proxy.
+We expect to handle metadata in the HTTP Connect case using http 'X' headers on the Connect request.
+This means that the metadata can only be sent when establishing a KAS to Proxy tunnel.
+For the GRPC case we just update the interface to the KAS.
+In this case the metadata can be sent even during tunnel usage.
```

Comment:
An example of this sort of metadata would be helpful. I'm not sure what it would be? Error propagation?

Reply:
My thought is that it might be good to indicate things like the type of traffic being forwarded (eg, logging, port forward, proxy, etc.) It could also indicate something like who the traffic is on behalf of. (eg. user foo or system account bar). Even which configuration rule is causing the traffic to be sent to this proxy.